### PR TITLE
RavenDB-20149 User can escape characters in `Search` | Proper ranking calculation for Search.

### PR DESCRIPTION
**Pull request by maciejaszyk**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-20149 

### Additional description

I introduced the same behavior as in Lucene, removed old "tricks" of how we build queries, and simplified the whole code of `Search`. 

### Type of change

- Bug fix

### How risky is the change?

- Moderate 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- Tests have been added that prove the fix is effective or that the feature works

### Testing by RavenDB QA team

- This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Corax/Constants.cs [hunk 1]
```diff
@@ -7,7 +7,12 @@ namespace Corax
     public static class Constants
     {
         public const string NullValue = "NULL_VALUE";
+        public static ReadOnlySpan<char> NullValueSpan => NullValue;
```
Comment (arekpalinski) [inline src/Corax/Constants.cs]: I think that the usage of `=>` will cause the conversion every time you access `NullValueSpan`. Let's calculate it just once.

Comment (maciejaszyk) [reply src/Corax/Constants.cs]: Done, in the case of byte array this pattern is no-allocating, but unfortunately, it's only for bytes.

Comment (redknightlois) [reply src/Corax/Constants.cs]: That's correct, bytes only. You could though do that internally with bytes and the access is a casting to char. I do that for some tables in SIMD algorithms.

**Comment by redknightlois** [comment]
LGTM

**Review by redknightlois — APPROVED** [review]

**Pull request merged**